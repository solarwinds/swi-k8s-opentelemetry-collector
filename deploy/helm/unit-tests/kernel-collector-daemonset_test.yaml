# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for kernel-collector-daemonset
templates:
  - network/kernel-collector-daemonset.yaml
  - network/configmap.yaml
tests:
  - it: DaemonSet spec should match snapshot when ebpfNetworkMonitoring is enabled
    template: network/kernel-collector-daemonset.yaml
    set:
      ebpfNetworkMonitoring.enabled: true
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: DaemonSet should not be generated when ebpfNetworkMonitoring is disabled
    template: network/kernel-collector-daemonset.yaml
    set:
      ebpfNetworkMonitoring.enabled: false
    asserts:
      - hasDocuments:
          count: 0
