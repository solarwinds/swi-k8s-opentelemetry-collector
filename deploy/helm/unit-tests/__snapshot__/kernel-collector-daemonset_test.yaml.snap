DaemonSet spec should match snapshot when ebpfNetworkMonitoring is enabled:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - args:
          - --config-file=/etc/network-explorer/config.yaml
          - --disable-nomad-metadata
          - --warning
        env:
          - name: EBPF_NET_CLUSTER_NAME
            value: <CLUSTER_NAME>
          - name: EBPF_NET_DISABLE_HTTP_METRICS
            value: "false"
          - name: EBPF_NET_KERNEL_HEADERS_AUTO_FETCH
            value: "true"
          - name: EBPF_NET_INTAKE_HOST
            value: RELEASE-NAME-swo-k8s-collector-network-k8s-reducer
          - name: EBPF_NET_INTAKE_PORT
            value: "7000"
          - name: EBPF_NET_CRASH_METRIC_HOST
            value: RELEASE-NAME-swo-k8s-collector-metrics-collector
          - name: EBPF_NET_CRASH_METRIC_PORT
            value: "4317"
          - name: BCC_PROBE_SUFFIX
            value: <CLUSTER_NAME>
        image: solarwinds/opentelemetry-ebpf-kernel-collector:v0.10.3
        imagePullPolicy: IfNotPresent
        name: swi-kernel-collector
        resources:
          requests:
            memory: 50Mi
        securityContext:
          privileged: true
        volumeMounts:
          - mountPath: /hostfs/
            name: host-root
            readOnly: true
          - mountPath: /hostfs/var/cache
            name: host-var-cache
            readOnly: false
          - mountPath: /etc/network-explorer
            name: network-explorer-config
            readOnly: true
    dnsPolicy: ClusterFirstWithHostNet
    hostNetwork: true
    hostPID: true
    initContainers:
      - command:
          - sh
          - -c
          - until nc -zv $EBPF_NET_INTAKE_HOST $EBPF_NET_INTAKE_PORT; do echo "Waiting for reducer endpoint..."; sleep 5; done;
        env:
          - name: EBPF_NET_INTAKE_HOST
            value: RELEASE-NAME-swo-k8s-collector-network-k8s-reducer
          - name: EBPF_NET_INTAKE_PORT
            value: "7000"
        image: busybox:1.36.1
        imagePullPolicy: IfNotPresent
        name: wait-for-reducer
    nodeSelector:
      kubernetes.io/os: linux
    securityContext:
      fsGroup: 0
      runAsGroup: 0
      runAsUser: 0
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 30
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - name: network-explorer-config
        projected:
          sources:
            - configMap:
                items:
                  - key: config.yaml
                    path: config.yaml
                name: RELEASE-NAME-swo-k8s-collector-network-otel-collector-config
      - hostPath:
          path: /
          type: Directory
        name: host-root
      - hostPath:
          path: /var/cache
          type: DirectoryOrCreate
        name: host-var-cache
